### PR TITLE
Feature/add tools

### PR DESCRIPTION
**Pull request by rjgriffin42**

### CODE REVIEW

File: src/test/java/us/ihmc/commons/robotics/robotSide/RobotSideTest.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
+package us.ihmc.commons.robotics.robotSide;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.Test;
+
+public class RobotSideTest
+{
+
+	@Test
```
Comment (ds58) [inline src/test/java/us/ihmc/commons/robotics/robotSide/RobotSideTest.java:10]: I think there was a tab inserted here (and also below)

File: src/test/java/us/ihmc/commons/robotics/robotSide/SideDependentListTest.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+import java.util.Map;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class SideDependentListTest
+{
+
+	@Test
```
Comment (ds58) [inline src/test/java/us/ihmc/commons/robotics/robotSide/SideDependentListTest.java]: I think there was a tab inserted here (and also below)

**Review by ds58 — CHANGES_REQUESTED** [review]

**Comment by ds58** [comment]
Looking good to me

**Review by ds58 — APPROVED** [review]

**Review by james-p-foster — APPROVED** [review]

File: build.gradle.kts [hunk 3]
```diff
@@ -32,6 +32,15 @@ mainDependencies {
    api("us.ihmc:log-tools:0.6.3")
 }
 
+
```
Comment (calvertdw) [inline build.gradle.kts]:
```suggestion
```

**Comment by calvertdw** [comment]
The README needs to get updated to say something about the new robotics part.

**Comment by rjgriffin42** [comment]
I plan on doing that when I do the release once this PR is done.

**Comment by calvertdw** [comment]
Can you please do the README update in this PR? I would like to review that and have that happen along with these changes.

File: README.md [hunk 4]
```diff
@@ -31,8 +43,21 @@ In your build.gradle:
 - Epsilons for explicitness and convenience.
 - Functional exception handling including one-liner option.
 - Recycling and preallocated lists, deque.
+- Ring buffers and array sorters
 - Tools to increase threading safety.
 - Tools for defining time intervals.
+- Tools for workign with Angles and arrays
+- Tools for workign with deadbands.
```
Comment (calvertdw) [inline README.md]: typos here

**Review by calvertdw — APPROVED** [review]

**Pull request merged**